File: server/datasets/java_source-code/Multiple-A2/organic5_bubble_sort.java
/*
 * Bubble Sort - organic5
 * Approach: Pairwise scanning with a while (true) / break loop.
 */

import java.util.Arrays;

class BubbleSort {
    public static int[] bubbleSort(int[] arr) {
        while (true) {
            boolean swapped = false;
            for (int i = 0; i < arr.length - 1; i++) {
                int a = arr[i];
                int b = arr[i + 1];
                if (a > b) {
                    arr[i] = b;
                    arr[i + 1] = a;
                    swapped = true;
                }
            }
            if (!swapped) {
                break;
            }
        }
        return arr;
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90, 5, 77, 1};
        System.out.println("Output: " + Arrays.toString(bubbleSort(arr)));
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_a2/organic5_bubble_sort.py
"""
Bubble Sort - organic5
Approach: Pythonic style pairing adjacent elements with
enumerate(zip(...)) inside a while True / break loop.
"""


def bubble_sort(arr):
    while True:
        swapped = False
        for i, (a, b) in enumerate(zip(arr, arr[1:])):
            if a > b:
                arr[i], arr[i + 1] = b, a
                swapped = True
        if not swapped:
            break
    return arr


if __name__ == "__main__":
    ARR = [64, 34, 25, 12, 22, 11, 90, 5, 77, 1]
    print("Output:", bubble_sort(ARR))
